src/main/scala/org/scalameter/Executor.scala

```diff
@@ -1,18 +1,21 @@
 package org.scalameter
 
 
-import collection._
-import compat._
-import utils.{withGCNotification, Tree}
+import utils.Tree
 
 
 
 trait Executor {
 
+  def measurer: Measurer
+
   def run[T](setuptree: Tree[Setup[T]], reporter: Reporter, persistor: Persistor): Tree[CurveData] = {
+    val context = measurer.prepareContext(setuptree.context)
+
     for (setup <- setuptree) yield {
       val exec = Option(setup.customExecutor).getOrElse(this)
-      val cd = exec.runSetup(setup)
+      val _cd = exec.runSetup(setup.copy(context = setup.context ++ context))
```

Comment:
With the above changes, here you won't need to modify the context, nor clean it.